### PR TITLE
Fix dynamic module support bug https://github.com/firebase/firebase-android-sdk/issues/3860

### LINKED ISSUE #3860: firebase-dynamic-module-support's monolithic play-core lib dependency causes duplicate classes
### [REQUIRED] Step 2: Describe your environment

  * Android Studio version: 
Android Studio Dolphin | 2021.3.1 Beta 4
Build #AI-213.7172.25.2113.8718688, built on June 14, 2022
Runtime version: 11.0.13+0-b1751.21-8125866 x86_64
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o.
macOS 12.4
GC: G1 Young Generation, G1 Old Generation
Memory: 2048M
Cores: 12
Registry:
    external.system.auto.import.disabled=true
    ide.text.editor.with.preview.show.floating.toolbar=false
    ide.balloon.shadow.size=0

Non-Bundled Plugins:
    com.dengzii.plugin.adb (2022.1.22)
    com.chrisrm.idea.MaterialThemeUI (6.16.2)

  * Firebase Component: com.google.firebase:firebase-dynamic-module-support
  * Component version: 16.0.0-beta01

### [REQUIRED] Step 3: Describe the problem
`firebase-dynamic-module-support` has a dependency on `com.google.android.play:core` which causes Duplicate class error durig compilation when using the new non-monolithic `com.google.android.play:app-update` and `com.google.android.play:review` libraries.

```
+--- com.google.firebase:firebase-dynamic-module-support:16.0.0-beta01
|    +--- com.google.android.play:core:1.9.0 -> 1.10.3
|    +--- com.google.firebase:firebase-common:20.0.0 -> 20.1.1 (*)
|    \--- com.google.firebase:firebase-components:17.0.0 (*)
```

part of the error output
```
Duplicate class com.google.android.play.core.appupdate.AppUpdateInfo found in modules app-update-2.0.0-runtime (com.google.android.play:app-update:2.0.0) and core-1.10.3-runtime (com.google.android.play:core:1.10.3)
Duplicate class com.google.android.play.core.appupdate.AppUpdateManager found in modules app-update-2.0.0-runtime (com.google.android.play:app-update:2.0.0) and core-1.10.3-runtime (com.google.android.play:core:1.10.3)
Duplicate class com.google.android.play.core.appupdate.AppUpdateManagerFactory found in modules app-update-2.0.0-runtime (com.google.android.play:app-update:2.0.0) and core-1.10.3-runtime (com.google.android.play:core:1.10.3)
Duplicate class com.google.android.play.core.appupdate.AppUpdateOptions found in modules app-update-2.0.0-runtime (com.google.android.play:app-update:2.0.0) and core-1.10.3-runtime (com.google.android.play:core:1.10.3)
Duplicate class com.google.android.play.core.appupdate.AppUpdateOptions$Builder found in modules app-update-2.0.0-runtime (com.google.android.play:app-update:2.0.0) and core-1.10.3-runtime (com.google.android.play:core:1.10.3)
Duplicate class com.google.android.play.core.appupdate.testing.FakeAppUpdateManager found in modules app-update-2.0.0-runtime (com.google.android.play:app-update:2.0.0) and core-1.10.3-runtime (com.google.android.play:core:1.10.3)
```

#### Steps to reproduce:

import both `com.google.firebase:firebase-dynamic-module-support:16.0.0-beta01` and `com.google.android.play:app-update` and/or `com.google.android.play:review`

#### Relevant Code:
```
implmentation "com.google.firebase:firebase-dynamic-module-support:16.0.0-beta01"
implmentation "com.google.android.play:review:2.0.0"
implmentation "com.google.android.play:app-update:2.0.0"
```

### PR DESCRIPTION
**Pull request by VinayGuthal**
Move from monolithic to partitioned apis at https://developer.android.com/reference/com/google/android/play/core/release-notes#partitioned-apis

Fixes #3860

### CODE REVIEW

File: firebase-components/firebase-dynamic-module-support/firebase-dynamic-module-support.gradle [hunk 1]
```diff
@@ -45,5 +45,8 @@ android {
 dependencies {
     implementation project(':firebase-common')
     implementation project(':firebase-components')
-    implementation 'com.google.android.play:core:1.9.0'
+    implementation 'com.google.android.play:app-update:2.0.0'
+    implementation 'com.google.android.play:review:2.0.0'
+    implementation 'com.google.android.play:feature-delivery:2.0.0'
```
[the remark this answers is not in the document]
Comment (rlazo) [reply firebase-components/firebase-dynamic-module-support/firebase-dynamic-module-support.gradle:50]: +1

**Comment by rlazo** [comment]
Reference the issue in the description instead of the title so it can be tracked by github.

File: firebase-components/firebase-dynamic-module-support/firebase-dynamic-module-support.gradle (hunk 1 above)
[the remark this answers is not in the document]
Comment (VinayGuthal) [reply firebase-components/firebase-dynamic-module-support/firebase-dynamic-module-support.gradle:50]: oh it's actually 3 out of 4.

[the remark this answers is not in the document]
Comment (VinayGuthal) [reply firebase-components/firebase-dynamic-module-support/firebase-dynamic-module-support.gradle:50]: but it looks like only one is needed.

**Review by vkryachko — APPROVED** [review]
Can you sync your branch to current master(specifically till after https://github.com/firebase/firebase-android-sdk/pull/3871), so that tests actually run for this PR?

**Review by rlazo — APPROVED** [review]

File: firebase-components/firebase-dynamic-module-support/firebase-dynamic-module-support.gradle [hunk 2]
```diff
@@ -45,5 +45,6 @@ android {
 dependencies {
     implementation project(':firebase-common')
     implementation project(':firebase-components')
-    implementation 'com.google.android.play:core:1.9.0'
+    implementation 'com.google.android.play:feature-delivery:2.0.0'
+
```
Comment (rlazo) [inline firebase-components/firebase-dynamic-module-support/firebase-dynamic-module-support.gradle:49]: rm empty line

**Pull request merged**